### ISSUE TITLE
Scaffolded database fails OnModelCreating when table has index with included columns snake_case named

### ISSUE THREAD

**Issue by adopilot** [dotnet/efcore#23068]
DbContex created scaffolding from existing  database fails on method OnModelCreating in runtime if in database has tables with indexed which using columns named snake_cased.

Here is steps.
In sql create table with index 

```
USE [Test]
GO
SET ANSI_NULLS ON
GO
SET QUOTED_IDENTIFIER ON
GO
CREATE TABLE [dbo].[Table_test](
	[Id] [int] IDENTITY(1,1) NOT NULL,
	[col_int] [int] NULL,
	[col_date] [date] NULL,
	[col_float] [float] NULL,
 CONSTRAINT [PK_Table_test] PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
CREATE NONCLUSTERED INDEX [IX_Test_Tune] ON [dbo].[Table_test]
(
	[col_int] ASC
)
INCLUDE ( 	[col_date],
	[col_float]) WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, DROP_EXISTING = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
GO
```

Then in Visual studio in Pacgae Manager Console 

`Scaffold-DbContext "data source=.;initial catalog=Test;persist security info=True;user id=xx;password=xxx;" Microsoft.EntityFrameworkCore.SqlServer -OutputDir Data\Test -f
`
This will generate DbContext.
App with this DbContext will build and run, but at first call to database OnModelCreating 
e.g: 
`var test= await _testContext.TableTests.ToListAsync();`
App will fall into exception 
**Include property 'TableTest.col_date' not found.**

Investigating OnModelCreating  I fund that these lines cause exception
```
entity.HasIndex(e => e.ColInt, "IX_Test_Tune") 
                    .IncludeProperties(new[] { "col_date", "col_float" });
```

My test table is scaffolded like this:
```
 modelBuilder.Entity<TableTest>(entity =>
            {
                entity.ToTable("Table_test");

                entity.HasIndex(e => e.ColInt, "IX_Test_Tune") 
                    .IncludeProperties(new[] { "col_date", "col_float" });//This part where is falling 

                entity.Property(e => e.ColDate)
                    .HasColumnType("date")
                    .HasColumnName("col_date");

                entity.Property(e => e.ColFloat).HasColumnName("col_float");

                entity.Property(e => e.ColInt).HasColumnName("col_int");
            });
```

Annoying thing is that this happening in runtime after app is started, and exception message is so unclear to find what is wrong.
If you have large set of index like this in db you have to run over and over to find all wrong indexes, and comment them.
_If we could change the column names then we would not use the db-first approach_

EF Core version: 5.0.0-rc.2.20475.6
Microsoft.EntityFrameworkCore.Tools :5.0.0-rc.2.20475.6
Target framework: .NET 5.0 RC2
Operating system: Windows 10 x64
SQL Server version: Microsoft SQL Server 2012 (SP3-GDR) (KB4019092) - 11.0.6251.0 (X64)   
IDE: Visual Studio 2019  16.8.0 Preview 5.0

**Comment by ErikEJ** [comment]
Have you tried -UseDatabaseNames option?

**Comment by adopilot** [comment]
Thank you @ErikEJ scaffolding with UseDatabaseNames will make dbcontext that hasn't got a problem with indexes that include colons with snake_case but using this option makes every entity have an ugly snake case name.

I already scaffolded the datebase with these indexes before .Net5 RC2.
Im not sure that this problem was caused by the newest version or  because i downloaded your DBEF power tools.
I noticed that now names of the ententes are in plural did EFcore changed convention again or did your plugin is mixed with Scaffold-DbContext command.
                                                                                                 Thank you
